packages/react/src/components/Authenticator/SignUp/SignUp.tsx

```diff
@@ -118,25 +118,39 @@ SignUp.AliasControl = ({
   placeholder = label,
 }) => {
   const {
-    components: { Input, Label, Text, ErrorText },
+    components: { Input, Flex, Label, ErrorText },
   } = useAmplify('Authenticator.SignUp.Alias');
   const [_state] = useAuth();
-  const { validationError } = getActorContext(_state) as SignUpContext;
+  const { validationError, formValues } = getActorContext(
+    _state
+  ) as SignUpContext;
   const error = validationError[name];
 
   return (
-    <>
-      <Label>
-        <Text>{label}</Text>
-        <Input
-          name={name}
-          placeholder={placeholder}
-          required
-          type={authInputAttributes[name].type}
-        />
-      </Label>
+    <Flex direction="column">
+      <Label htmlFor={name}>{label}</Label>
+      {name === 'phone_number' && (
+        <select
+          aria-label="country code"
+          name="country_code"
+          defaultValue={formValues.country_code}
+        >
+          {countryDialCodes.map((dialCode) => (
+            <option key={dialCode} value={dialCode}>
+              {dialCode}
+            </option>
+          ))}
+        </select>
+      )}
```

Comment:
We should try using the SelectField primitive that we have instead.

https://github.com/aws-amplify/amplify-ui/blob/main/packages/react/src/primitives/SelectField/SelectField.tsx

Reply:
Good call, this will also be implemented as part of [PhoneNumberField](https://app.asana.com/0/1200881947262804/1200683483662286) primitive. I had mostly just implemented core functionality in this PR.